albumentations/augmentations/transforms.py

```diff
@@ -3177,3 +3178,135 @@ def get_transform_init_args_names(self):
+        self.shift_x = shift_x
+        self.shift_y = shift_y
+        self.random_offset = random_offset
+        self.fill_value = fill_value
+        assert 0 < self.ratio < 1, "ratio must be between 0 and 1."
+
+    def apply(self, image, fill_value=0, holes=[], **params):
+        return F.cutout(image, holes, fill_value)
+
+    def get_params_dependent_on_targets(self, params):
+        img = params["image"]
+        height, width = img.shape[:2]
+        # set grid using unit size limits
+        if self.unit_size_min and self.unit_size_max:
+            assert (
+                2 <= self.unit_size_min <= self.unit_size_max
+            ), "Max unit size should be >= min size, both at least 2 pixels."
+            assert self.unit_size_max <= min(height, width), "Grid size limits must be within the shortest image edge."
+            unit_width = random.randint(self.unit_size_min, self.unit_size_max + 1)
+            unit_height = unit_width
+        else:
+            # set grid using holes numbers
+            if self.holes_number_x is None:
+                unit_width = max(2, width // 10)
+            else:
+                assert (
+                    1 <= self.holes_number_x <= width // 2
+                ), "The hole_number_x must be between 1 and image width//2."
+                unit_width = width // self.holes_number_x
+            if self.holes_number_y is None:
+                unit_height = max(min(unit_width, height), 2)
+            else:
+                assert (
+                    1 <= self.holes_number_y <= height // 2
+                ), "The hole_number_y must be between 1 and image height//2."
+                unit_height = height // self.holes_number_y
+
+        hole_width = int(unit_width * self.ratio)
+        hole_height = int(unit_height * self.ratio)
+        # min 1 pixel and max unit length - 1
+        hole_width = min(max(hole_width, 1), unit_width - 1)
+        hole_height = min(max(hole_height, 1), unit_height - 1)
+        # set offset of the grid
+        if self.shift_x is None:
+            shift_x = 0
+        else:
+            shift_x = min(max(0, self.shift_x), unit_width - hole_width)
+        if self.shift_y is None:
+            shift_y = 0
+        else:
+            shift_y = min(max(0, self.shift_y), unit_height - hole_height)
+        if self.random_offset:
+            shift_x = random.randint(0, unit_width - hole_width)
+            shift_y = random.randint(0, unit_height - hole_height)
+        holes = []
+        for i in range(width // unit_width + 1):
+            for j in range(height // unit_height + 1):
+                x1 = min(shift_x + unit_width * i, width)
+                y1 = min(shift_y + unit_height * j, height)
+                x2 = min(x1 + hole_width, width)
+                y2 = min(y1 + hole_height, height)
+                holes.append((x1, y1, x2, y2))
+
+        return {"holes": holes}
+
+    @property
+    def targets_as_params(self):
+        return ["image"]
```

Comment:
I think you do not need this property, because `height` and `width` already in the params with names `rows` and `cols`

Reply:
did not know that, thank you

Reply:
Strange that params does not contains this values. Will need to check it out in other transforms, thanks for info.